File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/TSLException.java
package tr.gov.tubitak.uekae.esya.api.tsl;

public class TSLException extends Exception {
	public TSLException() {
	}

	public TSLException(String aMessage) {
		super(aMessage);
	}

	public TSLException(String aMessage, Throwable aCause) {
		super(aMessage, aCause);
	}

	public TSLException(Throwable aCause, String aMessage)

	{
		super(aMessage, aCause);
	}

	public TSLException(Throwable aCause)

	{
		super(aCause.getMessage(), aCause);
	}

}

File: api-tsl/csharp/src/TSLException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace tr.gov.tubitak.uekae.esya.api.tsl
{
    public class TSLException : Exception
    {
        public TSLException()
        {
        }

        public TSLException(String aMessage):base(aMessage)
        {            
        }

        public TSLException(String aMessage, Exception aCause):base(aMessage, aCause)
        {            
        }

        public TSLException(Exception aCause, String aMessage)
            : base(aMessage, aCause)
        {
        }       


        public TSLException(Exception aCause)
            : base(aCause.Message, aCause) //:))
        {            
        }
    }
}
